src/placeos-models/asset.cr

```diff
@@ -0,0 +1,54 @@
+require "./base/model"
+
+module PlaceOS::Model
+  class Asset < ModelBase
+    include RethinkORM::Timestamps
+
+    table :asset
+
+    attribute name : String, es_subfield: "keyword"
+    attribute category : String = ""
+    attribute description : String = ""
+    attribute size : String = ""
+
+    attribute purchase_date : Int32
+    attribute good_until_date : Int32?
+
+    attribute barcode : Int32
+    attribute brand : String = ""
+
+    # Saved as cents
+    attribute purchase_price : Int32
+
+    # Array of URLs to images for an asset
+    attribute images : Array(String) = [] of String
+
+    # URL of downloadable receipt
+    attribute invoice : String?
+
+    attribute quantity : Int32 = 0
+    attribute in_use : Int32 = 0
+
+    attribute other_data : JSON::Any = JSON::Any.new({} of String => JSON::Any), converter: JSON::Any::StringConverter, es_type: "object"
+
+    attribute consumable_assets : Array(Asset)? = [] of Asset
```

Comment:
To simplify deleting of assets, this may be better as a `has_many` relation.
Otherwise, on delete you'd need to ensure that every asset has the deleted asset removed from its `consumable_asset`

Reply:
The relation wouldn't be `destroy` though.

Reply:
```suggestion
```